app/controllers/oidc_controller.rb

```diff
@@ -3,17 +3,28 @@
 class OidcController < ApplicationController
   def callback
     reset_session
-    session[:auto_generated_data] = {
-      example_1: 'Lorem ipsum est...',
-      example_2: 'Something or other'
-    }
-    session[:userinfo] = request.env['omniauth.auth'].info
-    redirect_to address_url
-  end
+    sub = request.env['omniauth.auth']['extra']['raw_info']['sub']
```

Comment:
possibly could create a variable called `user_auth_info = request.env['omniauth.auth']` then you can reference the the sub as `user_auth_info.uid` and later assign `session[:userinfo] = user_auth_info.info` - it saves the need to access the request.env multiple times